src/chapters/setup-intro-data-types/alternate-code-practice.rst

```diff
@@ -0,0 +1,406 @@
+   variable. Your ``CLASSPATH`` environment variable can name two kinds of
+   structures:
+
+   a. A *Java archive file*, ``.jar``, that contains Java classes.
+   b. Another Unix directory that contains Java class files.
+
+Declaring Variables
+--------------------
+
+In the example above, lines 5 - 7 contain variable declarations:
+
+.. sourcecode:: java
+   :lineno-start: 5
+
+   double fahrenheit;
+   double celsius;
+   Scanner input;
+
+Since Java is a statically typed language, we must always declare the data type
+for any variable. Lines 5 & 6 establish that ``fahrenheit`` and ``celsius``
+will hold values of type ``double``. In line 7, the variable ``input``
+references a ``Scanner`` object.
+
+If later in the code we try to initialize ``fahrenheit`` with a string:
+
+.. sourcecode:: java
+
+   fahrenheit = "xyz"
+
+the compiler throws an error because ``fahrenheit`` is supposed to be a double.
+
+The following error is common for new Java programmers. Suppose we forget to
+include the declaration for ``celsius``. What happens if we try to
+manually compile our program using ``javac TempConverter.java`` on the command
+line?
+
+.. admonition:: Try It
+
+   #. Edit your ``TempConverter`` class by removing line 6, which declares the
+      variable ``celsius``.
+   #. Using your terminal, navigate to the parent directory of your
+      ``TempConverter.java`` class and run ``javac TempConverter.java``.
+
+Your terminal will return some errors that resemble these:
+
+.. sourcecode:: bash
+
+   Error:(14, 9) java: cannot find symbol
+   symbol:   variable celsius
+   location: class TempConverter
+
+   Error:(15, 64) java: cannot find symbol
+   symbol:   variable celsius
+   location: class TempConverter
+
+These two *compiler errors* occur before the program runs. The values in the
+parentheses ``()`` give the line number and text column where the error was
+found. In the first description (line 14, column 9), the ``celsius`` variable
+before the ``=`` is flagged. When this type of error happens, it usually means
+that the variable was not declared before we tried to initialize it with a
+value.
+
+The second error message (line 15, column 64) occurs because we use
+``celsius`` before it has been assigned a value.
+
+.. admonition:: Note
+
+   When using an IDE such as IntelliJ, your work is typically checked by the
+   IDEs built-in compiler as you write your code. Errors are often visually
+   indicated by the IDE as you type. This avoids having to explicitly
+   compile your code before finding errors. Nice, huh?
+
+   .. figure:: figures/IDE-flagged-errors.png
+      :alt: The ``celsius`` variables are flagged.
+
+      The red coloring of the ``celsius`` variables indicate errors.
+
+The general rule in Java is that you must decide what kind of object your
+variable is going to reference and then declare that variable before you use
+it. There is much more to say about the static typing of Java, but for now this
```

Comment:
"The general rule in Java is that you must decide what type your
variable references and then declare that variable before you use
it."

Reply:
Much better phrasing.  I tweaked the words just a little before pasting.  Thanks.